docs/book.md

```diff
@@ -0,0 +1,299 @@
+                                                # `userAddress`, so this is done to expose idiomatic Scala. The underlying
+                                                # marshallers and unmarshallers maintain this mapping for you though,
+                                                # so no chance of protocol violations.
+
+            $ref: '#/definitions/UserAddresses' # Ensures that tye type of `userAddress` will be `IndexedSeq[UserAddress]`
+
+      UserAddresses:
+        type: array
+
+        items:                                  # `items` is a special key for `type: array`, identifying the structure of the
+                                                # sequence members
+
+          $ref: '#/definitions/UserAddress'     # Primitive types could be listed here, but as we're referring to another class,
+                                                # we need to explicitly use a `$ref`. This may change in the future,
+                                                # see https://github.com/twilio/guardrail/issues/76
+
+      UserAddress:
+        type: object
+        properties:
+          line1:
+            type: string
+          line2:
+            type: string
+          line3:
+            type: string
+
+Generating a Server
+===================
+
+Guardrail-generated servers come in two parts: a `Resource` and a `Handler`. The `Resource` contains all the routing logic, and accepts a `Handler` as an argument to the `route` function in order to provide an HTTP service in whatever HTTP framework you're hosting your service in. The following is an example from the akka-http server generator:
+
+    trait UserHandler {
+      def createUser(respond: UserResource.createUserResponse.type)(body: User): scala.concurrent.Future[UserResource.createUserResponse]
+      def createUsersWithArrayInput(respond: UserResource.createUsersWithArrayInputResponse.type)(body: IndexedSeq[User]): scala.concurrent.Future[UserResource.createUsersWithArrayInputResponse]
+      def createUsersWithListInput(respond: UserResource.createUsersWithListInputResponse.type)(body: IndexedSeq[User]): scala.concurrent.Future[UserResource.createUsersWithListInputResponse]
+      def loginUser(respond: UserResource.loginUserResponse.type)(username: String, password: String): scala.concurrent.Future[UserResource.loginUserResponse]
+      def logoutUser(respond: UserResource.logoutUserResponse.type)(): scala.concurrent.Future[UserResource.logoutUserResponse]
+      def getUserByName(respond: UserResource.getUserByNameResponse.type)(username: String): scala.concurrent.Future[UserResource.getUserByNameResponse]
+      def updateUser(respond: UserResource.updateUserResponse.type)(username: String, body: User): scala.concurrent.Future[UserResource.updateUserResponse]
+      def deleteUser(respond: UserResource.deleteUserResponse.type)(username: String): scala.concurrent.Future[UserResource.deleteUserResponse]
+    }
+    object UserResource {
```

Comment:
Does this change if you use different HTTP frameworks?

Reply:
Good question, leaving a comment in the code.